Posts/2021/02/ScriptingGuy.0002.GetDateOfYesterday.md

```diff
@@ -0,0 +1,182 @@
+# Getting Yesterday's Date
+
+**Q:** How can I get yesterday's date?
+
+**A:** You can use a combination of the `Get-Date` cmdlet and .NET Time/Date methods.
+
+First, let's look at dates in PowerShell and .NET
+Then we can look at how to calculate yesterday and use that in your scripts.
+
+## Dates in PowerShell
+
+Let's start by looking at how you can deal with dates and times.
+As you probably know, PowerShell contains the `Get-Date` cmdlet.
+This cmdlet returns a .NET **System.DateTime** object containing a **System.DateTime** object.
+
+Using the `Get-Date` cmdlet, you can get any date and time, and either display it or store it in a variable. 
+To get today's date. you could do this:
+
+```powershell
+# Get the current date
+Get-Date
+# Store the date in a variable then view it
+$Now = Get-Date
+$Now
+```
+
+The output looks like this:
+
+```powershell-console
+PS C:\> # Get the current date
+PS C:\> Get-Date
+08 January 2021 11:24:46
+
+# Store the date in a variable
+$Now = Get-Date
+$Now
+08 January 2021 11:24:47
+```
+
+As mentioned, the `Get-Date` cmdlet returns an object whose type is **System.DateTime**.
+This .NET structure provides a rich set of properties and methods to help you manipulate the date/time object.
```

Comment:
Isn't `DateTime` an immutable object? Might clarify this that the methods return a new object and the properties only have getters.

Reply:
I have added a comment 'below' that addresses this good catch.